### PR TITLE
transfer of json2xlsx scripts

### PR DESCRIPTION
**Pull request by frheault**
Transfert of script from bitbucket to github, switch to python3, update requirements

### CODE REVIEW

File: scripts/scil_convert_json_to_xlsx.py [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+    if len(dataframes):
+        _write_dataframes(dataframes, df_names, xlsx_path)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+
+    p.add_argument('in_json',
+                   help='File containing the json stats')
```
Comment (arnaudbore) [inline scripts/scil_convert_json_to_xlsx.py:379]: . dots dots dots

File: scripts/scil_convert_json_to_xlsx.py [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, args.in_json)
+    assert_outputs_exist(parser, args, args.out_xlsx)
+
+    _create_xlsx_from_json(args.in_json, args.out_xlsx,
```
Comment (arnaudbore) [inline scripts/scil_convert_json_to_xlsx.py:410]: maybe remove all _ before the name of functions...These are not private or related to a class.
+ docstring for everybody !

File: scripts/scil_convert_json_to_xlsx.py [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+import argparse
+import json
+
+import numpy as np
+import pandas as pd
+
+from scilpy.io.utils import (add_overwrite_arg,
+                             assert_inputs_exist, assert_outputs_exist)
+
+DESCRIPTION = ''
```
Comment (arnaudbore) [inline scripts/scil_convert_json_to_xlsx.py:17]: not used

**Comment by arnaudbore** [comment]
not tested @mdesco can you test it ?

File: scripts/scil_convert_json_to_xlsx.py (hunk 2 above)
Comment (frheault) [reply scripts/scil_convert_json_to_xlsx.py:408]: These functions are quite cryptic, really specific, any chance the docstring could be done by the person who did them?

File: scripts/scil_convert_json_to_xlsx.py [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+    if len(dataframes):
+        _write_dataframes(dataframes, df_names, xlsx_path)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+
+    p.add_argument('in_json',
+                   help='File containing the json stats.')
```
Comment (mdesco) [inline scripts/scil_convert_json_to_xlsx.py:377]: put extension in () please. (.json)

File: scripts/scil_convert_json_to_xlsx.py [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+
+    p.add_argument('out_xlsx',
+                   help='Output Excel file for the stats.')
```
Comment (mdesco) [inline scripts/scil_convert_json_to_xlsx.py:380]: (.xlsx)

File: scripts/scil_convert_json_to_xlsx.py [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+
+    p.add_argument('--no_sort_subs', action='store_false',
+                   help='If set, subjects won\'t be sorted alphabetically.')
+
+    p.add_argument('--no_sort_bundles', action='store_false',
+                   help='If set, bundles won\'t be sorted alphabetically.')
+
+    p.add_argument('--ignore_bundles', metavar='FILE',
+                   help='Path to a text file containing a list of bundles '
+                        'to ignore.')
```
Comment (mdesco) [inline scripts/scil_convert_json_to_xlsx.py:390]: (.txt) 

One bundle per line? I never used this option? What is the format? Can you provide a bit more info

File: scripts/scil_merge_json.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+    return new_dict
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+
+    p.add_argument('in_json', nargs='+',
+                   help='List of json files to merge.')
```
Comment (mdesco) [inline scripts/scil_merge_json.py:45]: (*.json)

File: scripts/scil_merge_json.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    p.add_argument('out_json',
+                   help='Output json file.')
```
Comment (mdesco) [inline scripts/scil_merge_json.py:47]: (.json)

**Comment by frheault** [comment]
All fixed and tested with your data @mdesco

**Pull request merged**